### PR TITLE
Package logcli as rpm and deb.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This patch introduces native packages for Linux of `logcli`. We use `nfpm` since it has no dependencies and opens the door to migrate to goreleaser.

For starters only Debian and RPM packages for amd64 are supported.

Outstanding:
- [ ] ~Sign packages.~
- [x] Publish packages to GitHub releases.
- [x] Set proper version.
- [x] Run in release pipeline.
- [ ] ~Optionally test package installation.~

### CODE REVIEW

File: nfpm.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+section: "default"
+provides:
+- logcli 
+maintainer: "Grafana Labs <support@grafana.com>"
+description: |
+  LogCLI is the command-line interface to Loki.
+  It facilitates running LogQL queries against a Loki instance.
+vendor: "Grafana Labs Inc"
+homepage: "https://grafana.com/loki"
+license: "AGPL-3.0"
```
Comment (jeschkies) [inline nfpm.yaml:18]: Is this actually the logcli license?

Comment (cyriltovena) [reply nfpm.yaml:18]: cc @slim-bean 

I think so.

File: nfpm.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# nfpm example config file
+#
+# check https://nfpm.goreleaser.com/configuration for detailed usage
+#
+name: "logcli"
+arch: "amd64"
+platform: "linux"
+version: "v2.3.0"
```
Comment (cyriltovena) [inline nfpm.yaml:8]: 2.4 ?

Comment (jeschkies) [reply nfpm.yaml:8]: Ah, the PR is a little older.

File: nfpm.yaml (hunk 1 above)
Comment (jeschkies) [reply nfpm.yaml]: Ping @slim-bean

File: tools/nfpm.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: "logcli"
+arch: "amd64"
+platform: "linux"
+version: ${CIRCLE_TAG}
```
Comment (dannykopping) [inline tools/nfpm.yaml:4]: What is `CIRCLE_TAG` and where is it defined?

**Review by dannykopping — APPROVED** [review]
LGTM

File: tools/nfpm.yaml (hunk 3 above)
Comment (jeschkies) [reply tools/nfpm.yaml:4]: It's provided by the Circle CI job (https://circleci.com/docs/2.0/env-vars/#built-in-environment-variables). It's already used by `tools/release-note.md` and `tools/release`.

**Review by owen-d — APPROVED** [review]
LGTM. `logcli` isn't in our [clients](https://github.com/grafana/loki/tree/main/clients) dir, meaning it's not Apache2 currently.

**Pull request merged**